### ISSUE TITLE
Schedule tasks to new workers

### ISSUE THREAD

**Issue by jasondavindev** [prestodb/presto#16155]
I'm using presto version `0.245.1` deployed in kubernetes cluster. My setup uses a Horizontal pod autoscaler for scale presto workers when there are high workload.
When the HPA scale up the workers pods, running queries doesn't uses new entries workers and then doesn't enjoy the entries resources.
My question is if there are a way for task manager schedule tasks for new entry workers for queries being executed?

**Comment by agrawalreetika** [comment]
@jasondavindev The reason being Presto cannot use upscaled nodes for HASH stages for running queries on the cluster, it can use new nodes only for the SOURCE stage (which includes Table Scan and partial aggregation) of the already running query.
One of the possible solutions, I am aware to this problem is to specify the required number of workers before you run large queries. There is the query-manager.required-workers option at the cluster level, which wouldn't be useful when worker pods are scaling up b/w min and max limit of HPA as it requires cluster restart. Exposing query-manager.required-workers property as session property from presto side can give you a possibility of setting the minimum number of workers before running heavy queries on downscaled Presto clusters.

**Comment by jasondavindev** [comment]
But if I set property for define minimum workers how the new worker will be scheduled since HPA metrics uses CPU threshold? The query remaning in wait state until that all workers are up?

**Comment by jasondavindev** [comment]
@agrawalreetika But if I set property for define minimum workers how the new worker will be scheduled since HPA metrics uses CPU threshold? The query remaning in wait state until that all workers are up?

**Comment by agrawalreetika** [comment]
@jasondavindev Yes if the query-manager.required-workers are not there in cluster query would be in WAITING_FOR_RESOURCES state.
One way could be in your HPA scaleup policy you can also have scale-up based on WAITING_FOR_RESOURCES query count in a cluster along with CPU

**Comment by jasondavindev** [comment]
Thanks for suggestion. I'll test it
